spec/p2p/v0.34/switch.md

```diff
@@ -0,0 +1,239 @@
+# Switch
+
+The `Switch` is documented in the [code](https://github.com/tendermint/tendermint/blob/46badfabd9d5491c78283a0ecdeb695e21785508/p2p/switch.go#L69) as follows:
+
+> // Switch handles peer connections and exposes an API to receive incoming messages
+> // on `Reactors`.
+
+The switch is created from a configuration and a `Transport` implementation.
+
+The switch implements a `BaseService` from `libs/service` package.
+
+## Adddress book
+
+`AddrBook` interface:
+
+> // An AddrBook represents an address book from the pex package, which is used
+> // to store peer addresses.
+
+TODO: document the use of the address book both in the switch and in the PEX reactor.
+
+## AddReactor
+
+The reactor is associated to the set of channel ids it employs.
+Two reactors (in the same node) cannot share the same channel id.
+
+There is a call back to the reactor, in which the switch passes itself to the
+reactor.
+
+For some reason, the added reactor is returned by this method.
+
```

Comment:
I think the return value is used only in tests. But there is a function in node.go to create a Custom reactor which calls `AddReactor`. But it is not used anywhere.

Reply:
This comment is not really relevant them.

Reply:
I've removed it.